workers/unity/Packages/io.improbable.gdk.buildsystem/Util/CommandlineParser.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.BuildSystem.Configuration;
+using Improbable.Gdk.Core;
+using UnityEditor;
+using UnityEditor.Build;
+
+namespace Improbable.Gdk.BuildSystem
+{
+    public static class CommandlineParser
+    {
+        public static string[] GetWorkerTypesToBuild(CommandLineArgs args)
+        {
+            var workerTypesArg = args.GetCommandLineValue("buildWorkerTypes", "UnityClient,UnityGameLogic");
+            var wantedWorkerTypes = workerTypesArg.Split(',');
+            return wantedWorkerTypes;
+        }
+
+        public static List<BuildTarget> GetBuildTargetFilter(CommandLineArgs args)
+        {
+            var buildTargetFilterArg = string.Empty;
+            if (!args.TryGetCommandLineValue("buildTargetFilter", ref buildTargetFilterArg))
+            {
+                return null;
+            }
+
+            return buildTargetFilterArg
+                .Split(',')
+                .Select(target =>
+                {
+                    switch (buildTargetFilterArg.ToLower())
+                    {
+                        case "android":
+                            return BuildTarget.Android;
+                        case "ios":
+                            return BuildTarget.iOS;
+                        case "winx86":
+                            return BuildTarget.StandaloneWindows;
+                        case "win":
+                            return BuildTarget.StandaloneWindows64;
+                        case "linux":
+                            return BuildTarget.StandaloneLinux64;
+                        case "macos":
+                            return BuildTarget.StandaloneOSX;
+                        default:
+                            throw new BuildFailedException(
```

Comment:
Think these should be `ArgumentOutOfRangeException`s. Then the top level handler should wrap it in a `BuildFailedException`